Automattic/wp-calypso
# Multi-Domain: Allowing instant insertion of domains into the mini-cart v2

Related to 4847-gh-Automattic/dotcom-forge
This is a follow-up of the reverted code on https://github.com/Automattic/wp-calypso/pull/84269
Slack context: p1702324186992199-slack-CKZHG0QCR


## Proposed Changes

* We can now add domains into the mini-cart and get an instant return (and button release)
* Everything occurs async
* After waiting for the addProductsToCart using await we're updating their orders with the replaceProductsInCart as we did before.
* Delete states are also instantly
* Delete domains will block advance button until the deletion is complete

## Testing Instructions



* Here's the happy path test:


https://github.com/Automattic/wp-calypso/assets/1044309/cda11d10-6565-44f5-8f0f-b6d5eb0298dc


* Here's the worst scenario test where the domains aren't available (reason why we reverted it). As you can see, we add all of them, and as soon as we get a negative feedback, we remove each of them from the cart:


https://github.com/Automattic/wp-calypso/assets/1044309/e53fdf66-06a3-4b42-9349-bdedcf51963b



 
## Pre-merge Checklist
